lib/mongoose_push/metrics/telemetry_metrics.ex

```diff
@@ -10,10 +10,13 @@ defmodule MongoosePush.Metrics.TelemetryMetrics do
   def metrics do
     [
       # Summary is not yet supported in TelemetryMetricsPrometheus
-      Telemetry.Metrics.distribution("mongoose_push.notification.send.time",
-        buckets: [100, 250, 500, 1000],
+      Telemetry.Metrics.distribution(
+        "mongoose_push.notification.send.time.microsecond",
+        event_name: [:mongoose_push, :notification, :send],
+        measurement: :time,
+        buckets: [10, 25, 50, 100, 200, 500, 1000],
         tags: [:status, :service, :error_category, :error_reason],
-        unit: {:native, :second}
+        unit: {:native, :microsecond}
```

Comment:
`second` is recommended default for Prometheus in general. Is there a reason for using non-default unit?

Reply:
In all our installations (MongooseIM + MongoosePush) metrics showing a dalay of execution time of something were in microseconds. So the conistency is the reason here.

Reply:
Too bad, but fair enough ;)

Reply:
When setting seconds we'd need to adjust the bucket sizes to be really small. This way the results are more readable.

Reply:
@janciesla8818 This doesn't matter at all, this is just a unit. It doesn't matter if buckets are `[10, 25, 50, 100, 200, 500, 1000]` milliseconds or `[0.01, 0.025, 0.05, 0.1, 0.2, 0.5, 1.0]` seconds. Those are the same. On Grafana or whatever you use for displaying this, you choose display unit separately anyway. Here we only choose unit in which this will be saved to Prometheus and which needs to be selected in Grafana as "source unit", nothing else.
That's why "conistency" reason with other service makes sense, but functionally there is no difference.

Reply:
BTW, do we need to make the conversion from `native` to `microsecond`. As far as I can tell the event `[:mongoose_push, :notification, :send]` reports in `microsecond` already.

Reply:
IMHO, the measurements should be made in `native` (if it isn't I wound change that) and if that happens to be the same as `microsecond`, it doesn't matter, having explicit translation here is preferable.

Reply:
@janciesla8818  @rslota In case of this event, it's done with `:timer.tc` which returns the time in `microsecond` (http://erlang.org/doc/man/timer.html#tc-3). The `native` time unit can be different though. On my machine, it's in `nanosecond`.
I have a hunch that what happens now is converting a value given in **microsecond** from **_native_** to **_microsecond_** will give the wrong result (at least on those machines where `native` equals to `nanosecond`)

Reply:
Indeed, `:timer.tc` changes time unit explicitly to `microsecond`. What I normally see in projects is custom macro `timed` that is essentially a copy of `:timer.tc`, but without time conversion from native and without additional unnamed function (for better stack-traces). This has added benefit of having in project "the" function/macro fir timing, no need to guess which standard one could be used.
This is what I would do ^, but obviously if you plan on using `:timer.tc`, then you need to make sure you only use that, and remove then conversion for metrics, or leave `{:microsecond , :microsecond}` for explicit context.

Reply:
@rslota , @michalwski thanks for the comments. I'll have it removed. We'll have the unit in the name of the event, so putting additional `{:microsecond , :microsecond}` could be a bit redundant.